File: Sort and Search/SelectionSort.py
from nose.tools import assert_equal

def selectionSort(arr):
	for i in range(len(arr)):
		minIndex = i
		for j in range(i+1, len(arr)):
			if arr[j] < arr[minIndex]:
				minIndex = j
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	return arr

class Selection_Sort(object):
	def test(self, func):
		assert_equal(func([1, 7, 4, 2, 10, 8, 20, 3]), [1, 2, 3, 4, 7, 8, 10, 20])
		assert_equal(func([1, 2, 3, 4, 5]), [1, 2, 3, 4, 5])
		print('TESTS PASSED')

t = Selection_Sort()
t.test(selectionSort)

File: Sort and Search/SelectionSort.js
function selectionSort(array){
	var len = array.length;
	for(var i=0;i<len;i++){
		var max = 0;
		for(var j=0;j<len-i;j++){
			if(array[j]>array[max]){
				max = j;
			}
		}
		var temp = array[len-1-i];
		array[len-i-1] = array[max];
		array[max] = temp;
	}
	return array;
}

console.log(selectionSort([6,2,4,1,9]));
console.log(selectionSort([1,2,3,4]));
console.log(selectionSort([9,8,7,6,5,4,3,2,1]));